src/providers/mod.rs

```diff
@@ -89,16 +89,14 @@ impl<'s> Providers<'s> {
             self.default_provider
         };
 
-        match self.providers.get(resolved_provider_id) {
```

Comment:
I'm replacing these `match` expressions with combinators as I find them. I wasn't aware of the combinator stuff when I started this repo.